### ISSUE TITLE
Need a `dylint_attribute` proc-macro package

### ISSUE THREAD

**Issue by smoelius** [trailofbits/dylint#2043]
At a minimum, it should support something like:
```rust
#![dylint_allow("general", crate_wide_allow)]
```
Which should expand to:
```rust
#![cfg_attr(dylint_lib = "general", allow(crate_wide_allow))]
```
I think such an expansion should work (https://rustc-dev-guide.rust-lang.org/macro-expansion.html), but I would need to verify.

Also, it would be nice if the proc-macro could warn if `crate_wide_allow` was not a lint in the `general` library. But I don't know how hard it would be to perform that check.
